frontend/public/module/k8s/k8s.ts

```diff
@@ -38,8 +39,53 @@ export const groupVersionFor = (apiVersion: string) => ({
   version: apiVersion.split('/').length === 2 ? apiVersion.split('/')[1] : apiVersion,
 });
 
+const parseAPIVersion = (version: string) => {
+  const parsed = /^v(\d+)(?:(alpha|beta)(\d+))?$/.exec(version);
+  return parsed
+    ? { majorVersion: Number(parsed[1]), qualifier: parsed[2], minorVersion: Number(parsed[3]) }
+    : null;
+};
+
+export const crdVersionSort = (crdVersions: CRDVersion[]) => {
+  return crdVersions?.sort((v1, v2) => {
+    const v1Parsed = parseAPIVersion(v1.name);
+    const v2Parsed = parseAPIVersion(v2.name);
+    // First sort on major version with no qualifiers: v3 > v1
+    if (
+      v1Parsed.majorVersion !== v2Parsed.majorVersion &&
+      !v1Parsed.qualifier &&
+      !v2Parsed.qualifier
+    ) {
+      return v2Parsed.majorVersion - v1Parsed.majorVersion;
+    }
+    // Then sort on any version with no qualifier over a qualifier: v1 > v3alpha
+    if (_.isEmpty(v1Parsed.qualifier) !== _.isEmpty(v2Parsed.qualifier)) {
+      return v1Parsed.qualifier ? 1 : -1;
+    }
+    // Beta beats alpha: v1beta1 > v1alpha1
+    const isBetaV1 = v1Parsed.qualifier === 'beta';
+    const isBetaV2 = v2Parsed.qualifier === 'beta';
+    if (isBetaV1 !== isBetaV2) {
+      return isBetaV1 ? -1 : 1;
+    }
+    // Same qualifier, then numeric values win: v2beta2 > v1beta2
+    if (v1Parsed.majorVersion !== v2Parsed.majorVersion) {
+      return v2Parsed.majorVersion - v1Parsed.majorVersion;
+    }
+    // Finally compare minor version: v1beta2 > v1beta1
+    return v2Parsed.minorVersion - v1Parsed.minorVersion;
+  });
+};
+
+export const getLatestVersionForCRD = (crdVersions: CRDVersion[], legacyCRDVersion: string) => {
```

Comment:
I would pass the CRD instead of fields from the CRD (fewer parameters and cleaner API)

```suggestion
export const getLatestVersionForCRD = (crd: CustomResourceDefinitionKind): string => {
```